com.unity.visualeffectgraph/Editor/Compiler/VFXCodeGenerator.cs

```diff
@@ -362,10 +367,190 @@ static private void SubstituteMacros(StringBuilder builder)
             }
         }
 
+        internal static void BuildContextBlocks(VFXContext context, VFXContextCompiledData contextData,
```

Comment:
Lines 370-532 were made to break up and re-use some of the components of VFXCodeGenerator; but pretty much all of this should be consolidated into something simpler as there is a lot happening that is redundant.